subsys/bluetooth/shell/bt.c

```diff
@@ -1147,14 +1152,85 @@ static int cmd_scan_filter_clear_addr(const struct shell *sh, size_t argc,
 #endif /* CONFIG_BT_OBSERVER */
 
 #if defined(CONFIG_BT_BROADCASTER)
-static const struct bt_data ad_discov[] = {
-	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
-};
+static void ad_mode_parse(const struct shell *sh, size_t argc, char *argv[], bool *discov)
+{
+	for (size_t argn = 0; argn < argc; argn++) {
+		const char *arg = argv[argn];
+
+		if (!strcmp(arg, "discov")) {
+			*discov = true;
+			break;
+		} else if (!strcmp(arg, "non_discov")) {
+			*discov = false;
+			break;
+		}
+	}
+}
+
+static ssize_t ad_init(struct bt_data *data, size_t data_size, bool discoverable)
+{
+	static uint8_t ad_flags;
+	size_t len = 0;
+	int err;
+
+	/* Set BR/EDR Not Supported if LE-only device */
+	ad_flags = IS_ENABLED(CONFIG_BT_BREDR) ? 0 : BT_LE_AD_NO_BREDR;
+
+	if (discoverable) {
+		ad_flags |= BT_LE_AD_GENERAL;
+	}
+
+	if (ad_flags != 0) {
+		__ASSERT(data_size > len, "No space for AD_FLAGS");
+		data[len].type = BT_DATA_FLAGS;
+		data[len].data_len = sizeof(ad_flags);
+		data[len].data = &ad_flags;
+		len++;
+	}
+
+#if defined(CONFIG_BT_AUDIO)
+	/* Advertise RSi in discoverable mode only */
+	if (IS_ENABLED(CONFIG_BT_CSIS) && csis != NULL && discoverable) {
+		static uint8_t ad_rsi[BT_CSIS_RSI_SIZE];
+
+		/* A privacy-enabled Set Member should only advertise RSI values derived
+		 * from a SIRK that is exposed in encrypted form.
+		 */
+		if (IS_ENABLED(CONFIG_BT_PRIVACY) &&
+		    !IS_ENABLED(CONFIG_BT_CSIS_ENC_SIRK_SUPPORT)) {
+			shell_warn(ctx_shell, "unencrypted SIRK, excluding AD_RSI");
```

Comment:
Hmm, not really sure about this, as we are basically overwriting a "shall" based on a "should", i.e. 

> When using this profile [CSIP] over the LE transport, Set Members in the GAP Peripheral role shall include the RSI AD Type (defined in [1]) in the Advertising Data or Scan Response Data.

is being blocked by

> A privacy-enabled Set Member should only advertise RSI values derivedfrom a SIRK that is exposed in encrypted form.

I think it is OK to add the warning, but IMO we should still advertise the RSI.